src/bufr_interface.h

```diff
@@ -17,151 +17,158 @@
+   @param unit the integer to use as the fortran file unit.
+   @param filepath path to the file we want to open.
+   
+   @author Ronald Mclaren @date 2020-07-29
 */
   void open_f(int unit, const char* filepath);
 
 
-/** @author Ronald McLaren
-    @date 2020-07-29
+/**
+   Wraps fortran "close" function so we can close a Fortran file
+   from a C program.
 
-    @brief Wraps fortran "close" statement so we can close a Fortran file
-           from a C program.
+   @param unit the integer to use as the fortran file unit.
 
-    @param[in] unit - int: the integer to use as the fortran file unit
+   @author Ronald Mclaren @date 2020-07-29
 */
   void close_f(int unit);
 
 
-/** @author Ronald McLaren
-    @date 2020-07-29
+/**
+   Wraps BUFRLIB openbf() subroutine.
 
-    @brief Wraps BUFRLIB "openbf" subroutine.
+   @param bufr_unit the fortran file unit number.
+   @param cio cio string (ex "IN", "SEC3", and "OUT").
+   @param table_file_id table_file unit number.
 
-    @param[in] bufr_unit - int: the fortran file unit number
-    @param[in] cio - const char*: cio string (ex "IN", "SEC3", and "OUT")
-    @param[in] table_file_id - int: table_file unit number
+   @author Ronald Mclaren @date 2020-07-29
 */
   void openbf_f(int bufr_unit, const char* cio, int table_file_id);
 
 
-/** @author Ronald McLaren
-    @date 2020-07-29
+/**
+   Wraps BUFRLIB closbf() subroutine.
 
-    @brief Wraps BUFRLIB "closbf" subroutine.
+   @param bufr_unit the fortran file unit number to close.
 
-    @param[in] bufr_unit - int: the fortran file unit number to close
+   @author Ronald Mclaren @date 2020-07-29
 */
   void closbf_f(int bufr_unit);
 
 
-/**  @author Ronald McLaren
-     @date 2020-07-29
+/**
+   Wraps BUFRLIB exitbufr() subroutine. Closes all open file units
+   used by BUFRLIB.
 
-     @brief Wraps BUFRLIB "exitbufr" subroutine. Closes
-            all open file units used by BUFRLIB.
+   @author Ronald Mclaren @date 2020-07-29
 */
   void exitbufr_f();
 
 
-/** @author Ronald McLaren
-    @date 2020-07-29
+/**
+   Wraps BUFRLIB ireadmg() subroutine.
 
-    @brief Wraps BUFRLIB "ireadmg" subroutine.
+   @param bufr_unit the fortran file unit number to read from.
+   @param subset the subset string.
+   @param iddate datetime of message.
+   @param subset_len length of the subset string.
 
-    @param[in] bufr_unit - int: the fortran file unit number to read from
-    @param[out] subset - char*: the subset string
-    @param[out] iddate - int*: datetime of message
-    @param[in] subset_len - int: length of the subset string
+   @return ???
+
+   @author Ronald Mclaren @date 2020-07-29
 */
   int ireadmg_f(int bufr_unit, char* subset, int* iddate, int subset_len);
 
 
-/** @author Ronald McLaren
-    @date 2020-07-29
+/**
+   Wraps BUFRLIB ireadsb() function.
+
+   @param bufr_unit the fortran file unit number to read from.
 
-    @brief Wraps BUFRLIB "ireadsb" function.
+   @return ???
 
```

Comment:
same as the underlying ireadsb() routine:

- 0 = new BUFR data subset was successfully read into internal arrays
- -1 = there are no more BUFR data subsets in the BUFR message

Reply:
Fixed.

Reply:
Actually, it looks like you copied the same text from ireadmg here.  I'll just push up another commit to fix this and add in some of the other requested updates...

Reply:
done